packages/rocketchat-videobridge/server/methods/jitsiGenerateToken.js

```diff
@@ -0,0 +1,57 @@
+import { Meteor } from 'meteor/meteor';
+import { RocketChat } from 'meteor/rocketchat:lib';
+import { jws } from 'jsrsasign';
+
+Meteor.methods({
+	'jitsi:generateAccessToken': () => {
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'jitsi:generateToken' });
+		}
+
+		const jitsiDomain = RocketChat.settings.get('Jitsi_Domain');
+		const jitsiApplicationId = RocketChat.settings.get('Jitsi_Application_ID');
+		const jitsiApplicationSecret = RocketChat.settings.get('Jitsi_Application_Secret');
+
+		function addUserContextToPayload(payload) {
+			const user = Meteor.user();
+			payload.context = {
+				user: {
+					name: user.name,
+					email: user.emails[0].address,
+					avatar: Meteor.absoluteUrl(`avatar/${ user.username }`),
+					id: user._id,
+				},
+			};
+			return payload;
+		}
+
+		const JITSI_OPTIONS = {
+			jitsi_domain: jitsiDomain,
+			jitsi_lifetime_token: '1hour', // only 1 hour (for security reasons)
+			jitsi_application_id: jitsiApplicationId,
+			jitsi_application_secret: jitsiApplicationSecret,
+		};
+
+		const HEADER = {
+			typ: 'JWT',
+			alg: 'HS256',
+		};
+
+		const commonPayload = {
+			iss: JITSI_OPTIONS.jitsi_application_id,
+			sub: JITSI_OPTIONS.jitsi_domain,
+			iat: jws.IntDate.get('now'),
+			nbf: jws.IntDate.get('now'),
+			exp: jws.IntDate.get(`now + ${ JITSI_OPTIONS.jitsi_lifetime_token }`),
+			aud: 'RocketChat',
+			room: '*',
```

Comment:
For additional access control to rooms maybe here we could pass the room?  That way users can only access meetings that Rocket.Chat allows access to.

Reply:
So maybe pass in this method then check to see if have access to the room.  If so generate the token for the room?  

Just a thought on how to apply some Access Controls to prevent issues like: https://github.com/RocketChat/Rocket.Chat/issues/13511 from being that big of a deal?

Reply:
I agree with @geekgonecrazy it will be necessary if not, all authenticated users can access to others rooms.  It may be a security issue.

Reply:
@geekgonecrazy you want to say that perhaps simply by passing the variable `jitsiRoom` here we can make the current session completely closed? it needs to be checked.

Reply:
there we go check latest commit.  :)